can/io/trc.py

```diff
@@ -211,12 +214,12 @@ def __init__(
         """
         super().__init__(file, mode="w")
         self.channel = channel
-        if type(file) is str:
+        if isinstance(file, str):
             self.filepath = os.path.abspath(file)
-        elif type(file) is TextIOWrapper:
+        elif isinstance(file, TextIOWrapper):
             self.filepath = "Unknown"
             logger.warning("TRCWriter: Text mode io can result in wrong line endings")
-            logger.debug(
+            logger.debug(  # pylint: disable=logging-fstring-interpolation
                 f"TRCWriter: Text mode io line ending setting: {file.newlines}"
             )
```

Comment:
Is there a reason for not doing:

```suggestion
            logger.debug(
                "TRCWriter: Text mode io line ending setting: %s",
                file.newlines,
            )
```

Reply:
The problem is: `file.newlines` returns either `None`, a string or a tuple of strings. So to make this robust, we'd have to convert None and the tuple to a string `str(file.newlines)`. But then there is no benefit in doing lazy interpolation.

Reply:
To be honest, the logic does not make sense at all. file.newlines will always return None here. If the author wants to set specific newline characters, it can be done with the [reconfigure](https://docs.python.org/3/library/io.html#io.TextIOWrapper.reconfigure) method.

A bit further down there is this, which is also incorrect:
```
    def _write_line(self, line: str) -> None:
        self.file.write(line + "\r\n")
```
I will try to improve this